src/core/mac/mac.cpp

```diff
@@ -461,7 +461,7 @@ ThreadError Mac::SetNetworkName(const char *aNetworkName)
     VerifyOrExit(strlen(aNetworkName) <= OT_NETWORK_NAME_MAX_SIZE, error = kThreadError_InvalidArgs);
 
     memset(&mNetworkName, 0, sizeof(mNetworkName));
-    strncpy(mNetworkName.m8, aNetworkName, sizeof(mNetworkName));
+    strncpy(mNetworkName.m8, aNetworkName, sizeof(mNetworkName) - 1);
```

Comment:
Just FYI, memset above is redundant since strncpy will fill the reset of the mNetworkName memory that is left before the end of sizeof(mNetworkName) with zeros.